### PR TITLE
Fixes for Sabre SDK

### PR DESCRIPTION
**Pull request by ltseeley**
Fixes some of the addressing methods and the `into_transaction_builder` method.

### CODE REVIEW

File: sdks/rust/src/protocol/payload.rs [hunk 1]
```diff
@@ -2342,7 +2342,11 @@ impl SabrePayloadBuilder {
 
         Ok(TransactionBuilder::new()
             .with_family_name("sabre".into())
-            .with_family_version(env!("CARGO_PKG_VERSION").into())
+            .with_family_version(format!(
+                "{}.{}",
+                env!("CARGO_PKG_VERSION_MAJOR"),
+                env!("CARGO_PKG_VERSION_MINOR")
+            ))
```
Comment (vaporos) [inline sdks/rust/src/protocol/payload.rs]: family version is the protocol the smart contract adheres to and should only really change when it is incompatible with previous versions.

File: sdks/rust/src/protocol/mod.rs [hunk 2]
```diff
@@ -47,7 +47,7 @@ pub fn compute_namespace_registry_address(namespace: &str) -> Result<Vec<u8>, Ad
         }
     };
     let hash = sha512_hash(prefix.as_bytes());
-    Ok([&parse_hex(NAMESPACE_REGISTRY_ADDRESS_PREFIX)?, &hash[..64]].concat())
+    Ok([&parse_hex(NAMESPACE_REGISTRY_ADDRESS_PREFIX)?, &hash[..32]].concat())
```
Comment (peterschwarz) [inline sdks/rust/src/protocol/mod.rs]: Could these prefix's be just byte array constants?

File: sdks/rust/src/protocol/payload.rs [hunk 3]
```diff
@@ -2295,7 +2295,7 @@ impl SabrePayloadBuilder {
             }) => {
                 let addresses = vec![
                     compute_contract_registry_address(&name)?,
-                    ADMINISTRATORS_SETTING_ADDRESS.into(),
+                    parse_hex(ADMINISTRATORS_SETTING_ADDRESS)?,
```
Comment (peterschwarz) [inline sdks/rust/src/protocol/payload.rs]: Anyway that this could be byte array constant?

File: cli/Dockerfile-installed-bionic [hunk 4]
```diff
@@ -49,7 +49,7 @@ COPY . /project
 WORKDIR /project/cli
 
 RUN export VERSION=$(../bin/get_version) \
- && sed -i -e s/version.*$/version\ =\ \"${VERSION}\"/ Cargo.toml \
+ && sed -i -e "0,/version.*$/ s/version.*$/version\ =\ \"$(../bin/get_version)\"/" Cargo.toml \
```
Comment (vaporos) [inline cli/Dockerfile-installed-bionic]: Why is replacing $VERSION here necessary?

File: sdks/rust/src/protocol/payload.rs (hunk 1 above)
Comment (ltseeley) [reply sdks/rust/src/protocol/payload.rs]: Isn't that implied by a major/minor version change? Not sure what change you are requesting.

File: cli/Dockerfile-installed-bionic (hunk 4 above)
Comment (ltseeley) [reply cli/Dockerfile-installed-bionic]: The build was failing and this was @rbuysse's fix; he should have more information.

File: sdks/rust/src/protocol/mod.rs (hunk 2 above)
Comment (ltseeley) [reply sdks/rust/src/protocol/mod.rs]: They can, but would require a new minor version release.

Comment (ltseeley) [reply sdks/rust/src/protocol/mod.rs]: Added the prefixes as byte arrays and used them where possible.

File: sdks/rust/src/protocol/payload.rs (hunk 3 above)
Comment (ltseeley) [reply sdks/rust/src/protocol/payload.rs]: Added the address as a byte array and used it here.

File: sdks/rust/src/protocol/payload.rs (hunk 1 above)
Comment (vaporos) [reply sdks/rust/src/protocol/payload.rs]: The family version isn't the same as the package version (of sabre), it's more like an API version of the smart contract itself.

**Pull request merged**